src/main.py

```diff
@@ -177,44 +171,60 @@ async def image(interaction: Interaction) -> None:
 @application_checks.has_role(config.dj_role_name)
 async def image_upload(interaction: Interaction, image_file: Attachment) -> None:
     """Upload a Google Forms image as attachment."""
-    global loaded_image
     # TODO: Some basic validity filtering
-    loaded_image.set(image_file.url)
+    # Persist the image URL
+    if not await persistent_state.save_form_image_url(interaction, image_file.url):
+        return
+
+    # Now load it again to ensure that it has saved correctly
+    loaded_image_url = persistent_state.get_form_image_url(interaction)
```

Comment:
In what sense does this "ensure that it has saved correctly" ? It seems more like you're loading it just to have a more readable variable name, and calling `get_form_image_url()` instead of using `image_file.url` for better logical separation of parts.

I guess maybe if the command throws an exception then something went wrong, but that feels like something that should happen in the save function

Reply:
Well, this is what we had before already with the call to `loaded_image.get()`.

It's essentially an automated call to `/image view`, which we currently always do after setting the image - just in case. But I agree that technically it shouldn't be necessary, as `save_form_image_url` should fail if the image wasn't set correctly.

We should really just trust that `/image` commands work if they don't fail and avoid the extra CPU cycles.

Reply:
Removed the loading in c3cb3ec.